client/src/security/keyprovider.rs

```diff
@@ -77,25 +77,18 @@ impl KeyProvider {
     ///
     /// assert_eq!(key, expected, "Key does not match expected {:?}", key);
     /// ```
+    #[deprecated = "This constructor promotes misuse of long passphrases"]
```

Comment:
```suggestion
    #[deprecated = "this constructor promotes misuse of long passphrases"]
```

Nit: Error messages are lowercase by convention and this probably also applies here.

Does it actually make sense to deprecate something in a major version change? If someone is using it right now, they can also migrate away from it, so it shouldn't be a problem, right?

Reply:
Thanks, didn't know about lower case in error messages.
I was reluctant about removing these weird constructors from KeyProvider because I don't understand what else they are used for. I would remove a lot of stuff from stronghold. In this particular case it's possible to use it in a safe manner, but it's not obvious how. Migration only deals with snapshot file format, but KeyProvider might be used for other things, such as providing keys as the name suggests.

Reply:
Fair argument. I don't have a strong opinion on this. Deprecating the function might be a decent intermediate solution to deter usage but still allow existing users to continue using it.